Metrics docs (#2256)
kroxylicious/kroxylicious: docs/modules/monitoring/proc-operator-setting-log-levels.adoc

// file included in the following:
//
// assembly-proxy-monitoring.adoc

[id='proc-proxy-setting-log-levels-{context}']
= Setting log levels

[role="_abstract"]

You can independently control the logging level of both the Kroxylicious Operator and the Kroxylicious Proxy.

In both cases, logging levels are controlled using two environment variables:

* `KROXYLICIOUS_APP_LOG_LEVEL` controls the logging of the application (`io.kroxylicious` loggers). It defaults to `INFO`.
* `KROXYLICIOUS_ROOT_LOG_LEVEL` controls the logging level at the root. It defaults to `WARN`.

When trying to diagnose a problem, start first by raising the logging level of `KROXYLICIOUS_APP_LOG_LEVEL`.
If more detailed diagnostics are required, try raising the `KROXYLICIOUS_ROOT_LOG_LEVEL`. Both the proxy and operator
use Apache Log4J2 and use logging levels understood by it: `TRACE`, `DEBUG`, `INFO`, `WARN`, and `ERROR`.

WARNING: Running either the operator or the proxy at elevated logging levels can generate significant amount of logs, which may consume significant storage.
Minimise the length of time the system is run in this state.

== Overriding proxy logging levels

> To override the logging level of a proxy, apply the environment variable to the Kubernetes `Deployment` resource that corresponds to the proxy instance in question.

Reviewer: Procedure layout should be like this with short active instructions as numbered steps:
 https://github.com/kroxylicious/kroxylicious/blob/b42c082d58d1a02176ada3497879a676003a88eb/docs/modules/multi-tenancy/proc-multi-tenancy.adoc

> To override the logging level of a proxy, apply the environment variable to the Kubernetes `Deployment` resource that corresponds to the proxy instance in question.

Reviewer: ```suggestion
Apply the environment variable to the proxy's Kubernetes `Deployment` resource to override its logging level.
```
